src/modules/claim/services/claim.service.ts

```diff
@@ -444,9 +470,13 @@ export class ClaimService {
   /**
    * Get allowed roles to revoke by given DID
    * @param {String} revokerDid - revoker DID
+   * @param {String} namespace - revoker DID
    * @returns allowed roles to revoke
    */
-  public async rolesByRevoker(revokerDid: string): Promise<Role[]> {
+  public async rolesByRevoker(
+    revokerDid: string,
+    namespace?: string
```

Comment:
IMO, it would be simpler if the filtering based on the namespace was done in `getByRevoker` instead of making `rolesByRevoker` more complex. In general, I believe it is best to reduce the arity of functions where possible, especially with regards to optional parameters. 
```suggestion
```

Reply:
changed this as per feedback